src/bot/integrations/tipeeestream.ts

```diff
@@ -0,0 +1,182 @@
+        reconnection: true,
+        reconnectionDelay: 1000,
+        reconnectionDelayMax: 5000,
+        reconnectionAttempts: Infinity,
+        query: {
+          access_token: this.apiKey.trim(),
+        },
+      });
+
+    if (this.socketToTipeeestream !== null) {
+      this.socketToTipeeestream.on('connect_error', (e) => {
+        error(chalk.red('TIPEEESTREAM.COM:') + ' error while connecting, ' + e);
+      });
+      this.socketToTipeeestream.on('connect', () => {
+        if (this.socketToTipeeestream !== null) {
+          this.socketToTipeeestream.emit('join-room', { room: this.apiKey.trim(), username: this.username.trim() });
+        }
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Successfully connected socket to service');
+      });
+      this.socketToTipeeestream.on('reconnect_attempt', () => {
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Trying to reconnect to service');
+      });
+      this.socketToTipeeestream.on('disconnect', () => {
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Socket disconnected from service');
+        this.disconnect();
+        this.socketToTipeeestream = null;
+      });
+
+      this.socketToTipeeestream.on('new-event', async (data) => {
+        if (data.event.type !== 'donation') {
+          return;
+        }
+        const { amount, message } = data.event.parameters;
+        const username = data.event.parameters.username.toLowerCase();
+        const donationCurrency = data.event.parameters.currency;
+
+        eventlist.add({
+          event: 'tip',
+          amount,
+          currency: donationCurrency,
+          username,
+          message,
+          timestamp: Date.now(),
+        });
+
+        events.fire('tip', {
+          username,
+          amount: parseFloat(amount).toFixed(2),
+          currency: donationCurrency,
+          amountInBotCurrency: Number(currency.exchange(amount, donationCurrency, currency.mainCurrency)).toFixed(2),
+          currencyInBot: currency.mainCurrency,
+          message,
+        });
+
+        alerts.trigger({
+          event: 'tips',
+          name: username,
+          amount: Number(parseFloat(amount).toFixed(2)),
+          currency: donationCurrency,
+          monthsName: '',
+          message,
+          autohost: false,
+        });
+
+        if (!data._is_test_alert) {
+          const getUser = async (username, id) => {
+            const userByUsername = await getRepository(User).findOne({ where: { username }});
+            if (userByUsername) {
+              return userByUsername;
+            }
+
+            const user = await getRepository(User).findOne({ where: { userId: id }});
+            if (user) {
+              return user;
+            } else {
+              return getRepository(User).save({
+                userId: Number(id),
+                username,
+              });
+            }
+          };
+
+          const user = await getUser(username, await users.getIdByName(username));
```

Comment:
Not a fan of those two parameters, I would move check if Id to getUser function, that way it would be called only if actually needed
```suggestion
          const user = await getUser(username);
```

Reply:
Right now getUser requires 2 params, so i cannot do `getUser(username)`